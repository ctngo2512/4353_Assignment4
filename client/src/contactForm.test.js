import React from 'react';
import fire from './fire';
import Hero from './Hero';
import ContactForm from './contactForm';
import {shallow, mount} from 'enzyme';
import {act, render} from '@testing-library/react';
import handleValidation from './contactForm';

window.alert = jest.fn();

describe('Contact Form component tests', ()=> {

    window.alert.mockClear();

    it("renders without crashing", () => {
        shallow(<Hero />);
    });

    it ('calls onSubmit prop function when form is submitted', () => {
        const wrapper = shallow(<ContactForm/>);
        expect(wrapper).toMatchSnapshot();
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
          });
    })

    it('Renders user data', () => {
        const fakeUser = {
            name: '',
            address: '',
            city: '',
            state: '',
            zipcode: ''
        }

        let snapshot = {val: () => fakeUser};
        jest.spyOn(fire, 'database').mockImplementation(() => ({
            ref: jest.fn().mockReturnThis(),
            on: jest.fn((event, callback) => callback(snapshot))
        }));
    
        act(() => {
            render(<ContactForm/>);
        });
    });
});