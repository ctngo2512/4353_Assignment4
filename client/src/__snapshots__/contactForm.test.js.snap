// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form component tests calls onSubmit prop function when form is submitted 1`] = `ShallowWrapper {}`;
